spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -759,10 +761,7 @@ private Table loadFromPathIdentifier(PathIdentifier ident) {
       return new SparkTable(table, snapshotIdAsOfTime, !cacheEnabled);
 
     } else if (branch != null) {
-      Snapshot branchSnapshot = table.snapshot(branch);
```

Comment:
Are removing this check because we plan to allow implicit branch creation in WAP?

Reply:
It's moved to the `SparkTable` constructor. In the WAP case, the table name will not have `.branch_xxx` suffix so won't hit this path.

Reply:
Makes sense, let me resolve this thread too.